src/ClientFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace oliverlorenz\reactphpmqtt;
+
+use oliverlorenz\reactphpmqtt\protocol\Version;
+use React\Dns\Resolver\Factory as DnsResolverFactory;
+use React\EventLoop\Factory as EventLoopFactory;
+
+class ClientFactory
+{
+    public static function createClient(Version $version, $resolverIp)
```

Comment:
I put that also to the roadmap. Like that as the first iteration improvement :)
Thank you for that :)